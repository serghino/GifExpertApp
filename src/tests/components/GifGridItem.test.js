import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Test component <GifGridItem />', () => {
    const title= 'Sergio';
    const url = 'http://test';
    const wrapper = shallow( <GifGridItem title={title} url={url}/> );
    
    test('should show <GifGridItem /> correctly', () => {
        expect( wrapper ).toMatchSnapshot();

    });
    test('should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    })
    
    test('should have an image with the title and url', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('should contain the class animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toContain('animate__fadeIn');
    })
    
})
