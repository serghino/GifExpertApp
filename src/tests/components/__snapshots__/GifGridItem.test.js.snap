// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifGridItem /> should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Sergio"
    src="http://test"
  />
  <p>
     
    Sergio
     
  </p>
</div>
`;
